### ISSUE TITLE
10000 files limit

### ISSUE THREAD

**Issue by bcnx** [aria2/aria2#1268]
Hi,

I'm using aria2 version 1.19.0 and a particular website I try to copy contains a lot of files (about 600000).
When I run aria2c, nomatter what distribution, it fills a folder with 10000 files and then stops adding files. 
It reports:

-> [RequestGroup.cc:758] errorCode=14 File renaming failed: download/index.html

Can I overcome this limitation? Would a more recent version solve this?

Cheers,

BC

**Comment by devvratmiglani** [comment]
Yeah, I kind of know why this happens, it's because the filename contains a reserved character that cannot be set as the file name in the operating system. You first have to sanitize the file names then download the file with the sanitized names, idk if aria2 supports that. You may use python to sanitize file names
